lib/plantuml-block/extension.rb

```diff
@@ -0,0 +1,40 @@
+require 'asciidoctor/extensions' unless RUBY_ENGINE == 'opal'
+require 'securerandom'
+
+include ::Asciidoctor
+
+# An extension that processes the contents of a block
+# as a PlantUML using servlet.
+#
+# Usage
+#
+#[plantuml]
+#----
+#
+#Bob->Alice : hello
+#
+#----
+#
+class PlantumlBlock < Extensions::BlockProcessor
+  use_dsl
+
+  named :plantuml
+  contexts [:listing, :literal, :open]
+  parse_content_as :raw
+
+  def process(parent, reader, attrs)
+    content = reader.lines.join("\n")
+    id = SecureRandom.uuid
+
+    html = "
+    <img id='#{id}' src='"+content+"' />
```

Comment:
Putting the raw content into the src attribute of the image tag seems like it could cause problems. Perhaps it's better to escape single quotes and stick it into a JavaScript variable, then creating the img element dynamically in JavaScript using createElement. You could still use a placeholder element as a parent to mark the location of the image. In fact, you should add it as:

```
<div id="#{id}" class="imageblock">
<div class="content">
</div>
</div>
```

(please use double quotes for attributes to be consistent with Asciidoctor).